playbooks/portals-setup-following.yml

```diff
@@ -39,16 +39,28 @@
     - name: Check docker was setup
       ansible.builtin.command: docker ps
       register: docker_ps_result
-      ignore_errors: True
+      # Use failed_when: False instead of ignore_errors: True so that here isn't
+      # a red logs that are confusing to users.
+      #
+      # NOTE: failed_when: False forces docker_ps_result.failed to be False but
+      # any errors are still recording in docker_ps_result.stderr
+      failed_when: False
 
     - name: Check docker SDK for Python3 was installed
       ansible.builtin.command: python3 -m pip show docker
       register: docker_pip_result
-      ignore_errors: True
-
+      # Use failed_when: False instead of ignore_errors: True so that here isn't
+      # a red logs that are confusing to users.
+      #
+      # NOTE: failed_when: False forces docker_pip_result.failed to be False but
+      # any errors are still recording in docker_pip_result.stderr
+      failed_when: False
+
+    # If both of the previous tasks were successful and didn't return any
+    # errors, we want to try and stop any active docker containers.
     - name: Include stopping portal and other docker containers (if exist)
       include_tasks: tasks/portal-stop-and-docker-containers-stop.yml
-      when: ((docker_ps_result.failed | default(False)) or (docker_pip_result.failed | default(False))) != True
+      when: docker_ps_result.stderr == '' and docker_pip_result.stderr == ''
```

Comment:
It's safer to use exit code, because stderr could be empty, but command(s) might be failing.

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#return-rc